dev-db/clickhouse/clickhouse-9999.ebuild

```diff
@@ -0,0 +1,724 @@
+
+		if [[ "${internal}" =~ ^.*ON$ ]]; then
+			[[ "${src}" ]] && CLICKHOUSE_UNPACK_OR_REMOVE+=( "${component}	${src}	${use_flag}" )
+		else
+			CLICKHOUSE_UNPACK_OR_REMOVE+=( "${component}" )
+			dyn="${src/,static-libs(+)/}"
+			dyn="${dyn/,static-libs/}"
+			dyn="${dyn/static-libs(+)/}"
+			dyn="${dyn/static-libs/}"
+			dyn="${dyn/[]/}"
+
+			if [[ "${use_flag}" && "${src}" ]]; then
+				DEPEND_GEN+="		${use_flag}? ( ${src} )"
+				RDEPEND_GEN+="		${use_flag}? ( ${dyn} )"
+			elif [[ "${src}" ]]; then
+				DEPEND_GEN+="		${src}"
+				RDEPEND_GEN+="		${dyn}"
+			fi
+			DEPEND_GEN+=$'\n'
+			RDEPEND_GEN+=$'\n'
+		fi
+		if [[ "${activate}" ]] ; then
+			CLICKHOUSE_CMAKE_USE_ARGUMENTS+=( "${activate}	${use_flag}" )
+		fi
+	done
+}
+
+_clickhouse_setup_vars
+DEPEND="${DEPEND_BASE}
+	static? (
+${DEPEND_GEN}	)
+"
+RDEPEND="${RDEPEND_BASE}
+	!static? (
+${RDEPEND_GEN}	)
+"
+
+_clang_fullversion() {
+	local ver="$1"; shift
+	set -- $($(tc-getCPP "$@") -E -P - <<<"__clang_major__ __clang_minor__ __clang_patchlevel__")
+	eval echo "$ver"
+}
+
+clang-fullversion() {
+	_clang_fullversion '$1.$2.$3' "$@"
+}
+
+clang-version() {
+	_clang_fullversion '$1.$2' "$@"
+}
+
+clang-major-version() {
+	_clang_fullversion '$1' "$@"
+}
+
+clang-minor-version() {
+	_clang_fullversion '$2' "$@"
+}
+
+clang-micro-version() {
+	_clang_fullversion '$3' "$@"
+}
+
+pkg_setup() {
+	use llvm && llvm_pkg_setup
+}
+
+pkg_pretend() {
+	if ! use static; then
+		ewarn "Non static builds are not supported by upstream."
+	fi
+
+	if use static && use '!static-libs'; then
+		ewarn "Static linking is requested but static libraries disabled. "
+		ewarm "Statically linked shared libraries will be created."
+		ewarm "Make sure dependencies are compiled with -fPIC."
+	fi
+
+	if [[ $(tc-getCC) == clang ]]; then
+		if [[ $(clang-major-version) -lt 6 ]]; then
+			eerror "Compilation with clang older than 6.0 is not supported"
+			die "Too old clang found"
+		fi
+	elif use libcxx; then
+		eerror "Compilation with libcxx is only supported when using clang"
+		die "Use clang to build with libcxx"
+	elif [[ $(gcc-major-version) -lt 7 ]] && [[$(gcc-minor-version) -lt 2 ]]; then
+		eerror "Compilation with gcc older than 7.2 is not supported"
+		die "Too old gcc found"
+	fi
+}
+
+src_unpack() {
+	default_src_unpack
+
+	[[ "${PV}" == 9999 ]] && git-r3_src_unpack
+	cd "${S}/contrib" || die "failed to cd to contrib"
+
+	local unpack_or_remove line lib src use_flag
+
+	unpack_or_remove=( "${CLICKHOUSE_UNPACK_OR_REMOVE[@]}" )
+
+	unpack_or_remove+=(
+		"rapidjson	rapidjson-${RAPIDJSON_COMMIT}.tar.gz \
+			$(usex json "$(usex '!cpu_flags_x86_avx2' json '!cpu_flags_x86_avx2')" json)"
+		"simdjson	simdjson-${SIMDJSON_COMMIT}.tar.gz \
+			$(usex json "$(usex cpu_flags_x86_avx2 json cpu_flags_x86_avx2)" json)"
+	)
+
+	for line in "${unpack_or_remove[@]}"; do
+		read -r lib src use_flag <<< "${line}"
+
+		if test -z "${src}" || ([[ "${use_flag}" ]] && ! use "${use_flag}" ); then
```

Comment:
Ebuilds are written in bash, so `test` should be replaced with `[[ ... ]]`. `( ... )` is a subshell, not condition grouping operator.

Reply:
(...) is used here as a subshell so `use` is only called when `$use_flag` is not empty

Reply:
Use `{ ... };`

Reply:
Fixed